### ISSUE TITLE
Symbol imported from stub but marked as unknown

### ISSUE THREAD

**Issue by twoertwein** [microsoft/pyright#3411]
**Describe the bug**
`tz_convert_from_utc` is typed and publicly re-exported but pyright says it is an unknown import symbol.

**To Reproduce**
```sh
git clone --depth=1 git@github.com:pandas-dev/pandas.git
cd pandas
sed -i "s/reportGeneralTypeIssues = false/reportGeneralTypeIssues = true/g" pyproject.toml
pyright --skipunannotated pandas/tseries/frequencies.py
```

> pandas/tseries/frequencies.py:10:5 - error: "tz_convert_from_utc" is unknown import symbol (reportGeneralTypeIssues)

**Expected behavior**

`pandas/_libs/tslibs/__init__.py` imports `tz_convert_from_utc` from `pandas/_libs/tslibs/vectorized.pyi` and exports it publicly by including it in `__all__`.

**Comment by twoertwein** [comment]
Probably the same issue but maybe easier to reproduce:

```py
import numpy as np  # 1.22.3

np.lib.stride_tricks
```
> error: "stride_tricks" is not a known member of module (reportGeneralTypeIssues)
> error: Type of "stride_tricks" is unknown (reportUnknownMemberType)

even though `numpy/lib/__init__.pyi` explicitly re-exports `stride_tricks` using the `as` syntax (but it also declares an empty(?) `__all__` at the same time)

**Comment by erictraut** [comment]
`tz_convert_from_utc` is imported by `frequencies.py` from the submodule `pandas._libs_tslibs`. This submodule, in turn, imports the symbol from the submodule `pandas._libs.tslibs.vectorized`. The stub file `vectorized.pyi` doesn't provide a type declaration for `tz_convert_from_utc`. Therefore its type is unknown.

The `stride_tricks` issue is a bug in the numpy type stub. There's a misspelling in the line:
```python
    stride_tricks as stride_stricks,
```

Note the extra `s` in `stride_stricks`.

**Comment by twoertwein** [comment]
Thank you!

> The stub file `vectorized.pyi` doesn't provide a type declaration for `tz_convert_from_utc`. Therefore its type is unknown.

I don't understand why the type of `tz_convert_from_utc` isn't declared. It seems to be here:
https://github.com/pandas-dev/pandas/blob/c7f615b6b34be13809b3440fc806f0382132d56d/pandas/_libs/tslibs/vectorized.pyi#L37

**Comment by erictraut** [comment]
Yes, there is a bug here. It occurs if you have an older version of pandas installed in your current Python environment and then open the local (cloned) pandas project. Pyright's import resolver was preferring the `vectorized.pyi` stub file from the installed library over the local one in this case, and the older version of `vectorized.pyi` didn't declare the symbol `tz_convert_from_utc`. This will be fixed in the next release.

**Comment by twoertwein** [comment]
wow, thank you as always for your thorough analysis! Yeah, I forgot I also had an older pandas installed :)

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.243, which I just published. It will also be included in the next release of pylance.
